README-iso.md

```diff
@@ -0,0 +1,260 @@
+- `install.img` (see below)
+- yum repository for the product (or a local mirror) for boot files and
+  local repository
+- additional files from `./iso/$RELEASE/`
+
+### `./scripts/create-installimg.sh`
+
+Creates `install-$RELEASE.img` for input to `create-install-iso`, from:
+- yum repository for the product (or a local mirror)
+- a `packages.lst` file listing RPMs to be installed
+- additional files from `./installimg/$RELEASE/`
+
+### `./scripts/mirror-repos.sh`
+
+Creates a local mirror of a subset of an official set of repositories
+for a given XCP-ng version, suitable for building an installation ISO.
+This scripts excludes from the mirror:
+- source RPMs
+- development RPMs
+- debugging-symbols RPMs
+
+### configuration layers and package repositories
+
+Configuration layers are defined as a subdirectory of the `configs/`
+directory.  Commands are given a layer search path as
+`<base-config>[:<config-overlay>]* `.
+
+Standard layers are organized such two of the standard layers must be
+used:
+
+* The "version" layer (e.g. `8.2`) provides required files:
+  - `packages.lst` and `yum.conf.tmpl` used to create the `install.img`
+    filesystem
+  - `yumdl.conf.tmpl` used to download files for the RPM repository
+    included in the ISO
+
+* The "repo" layers (e.g. `updates`) each provide a yum repo
+  configuration file, and optionally an `INCLUDE` file to pull
+  additional base repo layers.  The `base` layer will always be in the
+  include chain.
+
+XCP-ng official repositories are located at
+https://updates.xcp-ng.org/ and most of them are available through
+standard "repo" layers; e.g. the `testing` repository for `8.2` LTS can
+be used as `8.2:testing`.
+
+Custom repositories can be added with `--define-repo` flag.  They will
+be used by `yum` using the first `CUSTOMREPO.tmpl` template found in
+the layer search path (one is provided in `base`).
+
+## examples
+
+### 8.3 updates and testing
+
+```
+./scripts/mirror-repos.sh 8.3
+
+sudo ./scripts/create-installimg.sh \
+    --srcurl file://$HOME/mirrors/xcpng/8.3 \
+    -o install-8.3-testing.img \
+    8.3:testing
+
+./scripts/create-install-iso.sh \
+    --srcurl file://$HOME/mirrors/xcpng/8.3 \
+    -V "XCP-NG_830_TEST" \
+    8.3:testing install-8.3-testing.img xcp-ng-8.3-testing.iso
+```
+
+### tip of 8.2 (8.2 + updates)
+
+```
+./scripts/mirror-repos.sh 8.2
+
+sudo ./scripts/create-installimg.sh \
+    --srcurl file://$HOME/mirrors/xcpng/8.2 \
+    8.2:updates
+
+./scripts/create-install-iso.sh \
+    --srcurl file://$HOME/mirrors/xcpng/8.2 \
+    -V "XCP-NG_82U_TEST" \
+    8.2:updates install-8.2.updates-x86_64.img xcp-ng-8.2.updates-install.iso
+```
+
+### testing boot modes in qemu
+
+PC BIOS:
+
+```
+qemu-system-x86_64 -serial stdio -m 1G -cdrom xcp-ng-8.3-install.iso
+```
+
+UEFI:
+
+```
+qemu-system-x86_64 -serial stdio -m 1G -cdrom xcp-ng-8.3-install.iso \
+  --bios /usr/share/edk2/ovmf/OVMF_CODE.fd -net none
+```
+
+### testing that scripts run correctly
```

Comment:
Can this be launched in a GH action?

Reply:
They do take some time, so we likely don't want to run the whole of them in GH.  To be discussed.